wallet/core/src/runtime/wallet.rs

```diff
@@ -725,10 +755,17 @@ impl Wallet {
     }
 
     pub async fn find_accounts_by_name_or_id(&self, pat: &str) -> Result<Vec<Arc<dyn Account>>> {
-        let accounts = self.active_accounts().inner().values().cloned().collect::<Vec<_>>();
+        let mut accounts = self.active_accounts().inner().values().cloned().collect::<Vec<_>>();
+        let legacy_accounts = self.legacy_accounts().inner().values().cloned().collect::<Vec<_>>();
+        accounts.extend(legacy_accounts);
```

Comment:
It can be done without allocating accounts and legacy accounts vectors.
Accounts_iter.chain(legacy_iter).filter().collect()

Reply:
Thanks for noting this. This was actually a bug (albeit benign). We recently changed the implementation pattern to have all legacy accounts be active and thus present in the `active_accounts` list. As such this iteration was not even necessary (old code that was benign in nature). Fixed in https://github.com/kaspanet/rusty-kaspa/pull/312/commits/abc0298866e878c5b303efdff4ab870303717e98